src/core/ext/filters/client_channel/lb_policy/xds/xds_override_host.cc

```diff
@@ -132,6 +140,8 @@ class XdsOverrideHostLb : public LoadBalancingPolicy {
 
     XdsOverrideHostLb* policy() { return policy_.get(); }
 
+    void detach() { key_ = absl::nullopt; }
+
    private:
     class ConnectivityStateWatcher : public ConnectivityStateWatcherInterface {
```

Comment:
Please take a look at a new implementation.

Note that `interested_parties` is recomputed on every invocation. It is done in case the watchers have dynamic `interested_parties`